source/Halibut.Tests/ResponseMessageCacheFixture.cs

```diff
@@ -0,0 +1,264 @@
+            {
+                var client = clientAndService.CreateClient<ICachingService>();
+
+                var result1 = client.CachableCall();
+                var result2 = client.AnotherCachableCall();
+
+                result1.Should().NotBe(result2);
+            }
+        }
+
+        [Test]
+        [TestCaseSource(nameof(ServiceConnectionTypeAndVersion))]
+        public async Task ForAServiceThatSupportsCaching_ResponseShouldBeDifferentForDifferentEndpoints(ServiceConnectionType serviceConnectionType, string halibutServiceVersion)
+        {
+            using var clientAndServiceOne = await CreateClientAndService(serviceConnectionType, halibutServiceVersion);
+            {
+                var clientOne = clientAndServiceOne.CreateClient<ICachingService>();
+
+                using var clientAndServiceTwo = await CreateClientAndService(serviceConnectionType, halibutServiceVersion);
+                var clientTwo = clientAndServiceTwo.CreateClient<ICachingService>();
+
+                var result1 = clientOne.CachableCall();
+                var result2 = clientTwo.CachableCall();
+
+                result1.Should().NotBe(result2);
+            }
+        }
+
+        [Test]
+        [TestCaseSource(nameof(ServiceConnectionTypeAndVersion))]
+        public async Task ForAServiceThatSupportsCaching_ResponseShouldBeDifferentForDifferentInputParameters(ServiceConnectionType serviceConnectionType, string halibutServiceVersion)
+        {
+            using var clientAndService = await CreateClientAndService(serviceConnectionType, halibutServiceVersion);
+            {
+                var client = clientAndService.CreateClient<ICachingService>();
+
+                var result1 = client.CachableCall(Guid.NewGuid());
+                var result2 = client.CachableCall(Guid.NewGuid());
+
+                result1.Should().NotBe(result2);
+            }
+        }
+
+        [Test]
+        [TestCaseSource(nameof(ServiceConnectionTypeAndVersion))]
+        public async Task ForAServiceThatSupportsCaching_ClientShouldBeAbleToForceSpecificErrorResponsesToBeCached(ServiceConnectionType serviceConnectionType, string halibutServiceVersion)
+        {
+            using var clientAndService = await CreateClientAndService(serviceConnectionType, halibutServiceVersion);
+            {
+                clientAndService.Octopus.OverrideErrorResponseMessageCaching = (response) =>
+                {
+                    if (response.Error.Message.Contains("CACHE ME"))
+                    {
+                        return true;
+                    }
+
+                    return false;
+                };
+
+                var client = clientAndService.CreateClient<ICachingService>();
+
+                Exception exception1 = null;
+                Exception exception2 = null;
+
+                TryCatch(() => client.CachableCallThatThrowsAnException($"UNCACHED"), e => exception1 = e);
```

Comment:
nit: I wonder if  a should/could throw assertion could be used here. I think that would mean you end up with:
```
var exception1 = Assert.shouldThrow(() => client.CachableCallThatThrowsAnException($"UNCACHED"))
```

Removing the need for the TryCatch method and removes the need for the null check.